### ISSUE TITLE
Hide notifications when the icon is clicked

### ISSUE THREAD

**Issue by sprite-1** [elementary/wingpanel-indicator-notifications#64]
When the bell icon is clicked, I wish it would hide all the open notifications because they just get in the way. I don't think it's necessary to show said notifications again after the notification menu is closed either because the user already demonstrated they know where to see them.

**Comment by peteruithoven** [comment]
I don't like this idea. I frequently leave notifications because I want to deal with them on a later date. (Even more on my Android phone)
Clearing the list is already possible by clicking the Clear all notifications button?

> the user already demonstrated they know where to see them.

Could you clarify what you mean here?

**Comment by sprite-1** [comment]
I **don't** mean clearing the notifications on this list
![image](https://user-images.githubusercontent.com/33685433/52905855-3bf83800-327b-11e9-8876-4666a8c7a0c3.png)

I mean't hiding the floating notifications because when the above menu is opened, those floating notifications cover it.

**Comment by peteruithoven** [comment]
Thanks for the clarification. I like your idea. 
I guess this would also be the only way to clear all floating notifications?

**Comment by sprite-1** [comment]
I was careful not to say "clear" because I don't actually want to make it have the same behavior as clicking the (X) button on the notification which dismisses it. I just wanted them to be hidden (i.e not shown on the desktop but can still be found in the notifications menu for interaction later).

Maybe that's another area of topic in itself but it's still unclear to me what the (X) button in floating notifications do. Sometimes they dismiss it completely, sometimes the notification stays on the menu.

**Comment by janxkoci** [comment]
> it's still unclear to me what the (X) button in floating notifications do. Sometimes they dismiss it completely, sometimes the notification stays on the menu

I think they should show up in the indicator but some fail to do so. Or they should be cleared and again, some fail to do so. Either way, it's buggy.

**Comment by danrabbit** [comment]
So what we need here is to create some kind of DBus API so that the notification server can find out if the notifications indicator is open.

We also have reports about not having other notifications overlapping indicators as well. So perhaps we should just make this a feature of the panel itself and have it apply for all indicators packed into the end portion of the panel.

@cassidyjames thoughts on that?

**Comment by cassidyjames** [comment]
@danrabbit yeah that seems more sane; just hide notifications (possibly pausing any of their timeouts) while end-side indicators are open.
